src/dune_engine/build_system.mli

```diff
@@ -68,17 +68,27 @@ module Progress : sig
     ; number_of_rules_executed : int
     }
 
+  val equal : t -> t -> bool
+
   val complete : t -> int
 
   val remaining : t -> int
 
   val is_determined : t -> bool
 end
 
-val get_current_progress : unit -> Progress.t
+module State : sig
+  type t =
+    | Initializing
+    | Building of Progress.t
+    | Interrupting_current_build
+    | Build_finished_with_success_and_waiting
+    | Build_finished_with_failure_and_waiting
```

Comment:
```suggestion
    | Restarting_current_build
    | Build_succeeded__now_waiting_for_changes
    | Build_failed__now_waiting_for_changes
```
I think "restarting" is much clearer than "interrupting". Also tried to make the two other variants a bit clearer (not 100% sure that I succeeded).

Reply:
The names are fine but why double underscores? Those are already used for name mangling

Reply:
As we discussed offline, double underscores here are meant to represent commas, as in "build failed, now waiting for changes". 

Looks like `Interrupting_current_build` isn't renamed to `Restarting_current_build` yet.

Reply:
Added the missing rename.